Spigot-API-Patches/0189-Add-Mob-Goal-API.patch

```diff
@@ -0,0 +1,341 @@
++    NamespacedKey POLAR_BEAR_HURT_BY_TARGET = NamespacedKey.minecraft("polar_bear_hurt_by_target");
++    NamespacedKey DOOR_OPEN = NamespacedKey.minecraft("door_open");
++    NamespacedKey RANDOM_TARGET_NON_TAMED = NamespacedKey.minecraft("random_target_non_tamed");
++    NamespacedKey ENDERMAN_A = NamespacedKey.minecraft("enderman_a");
++    NamespacedKey BOW_SHOOT = NamespacedKey.minecraft("bow_shoot");
++    NamespacedKey MONSTER_PATROLLING_A = NamespacedKey.minecraft("monster_patrolling_a");
++    NamespacedKey PERCH = NamespacedKey.minecraft("perch");
++    NamespacedKey AVOID_TARGET = NamespacedKey.minecraft("avoid_target");
++    NamespacedKey RANDOM_FLY = NamespacedKey.minecraft("random_fly");
++    NamespacedKey SLIME_RANDOM_JUMP = NamespacedKey.minecraft("slime_random_jump");
++    NamespacedKey FOX_GOTO_TARGET = NamespacedKey.minecraft("fox_goto_target");
++    NamespacedKey RANDOM_STROLL_LAND = NamespacedKey.minecraft("random_stroll_land");
++    NamespacedKey GHAST_ATTACK_TARGET = NamespacedKey.minecraft("ghast_attack_target");
++    NamespacedKey PANDA_PANIC = NamespacedKey.minecraft("panda_panic");
++    NamespacedKey FOX_B = NamespacedKey.minecraft("fox_b");
++    NamespacedKey PANDA_AVOID_TARGET = NamespacedKey.minecraft("panda_avoid_target");
++    NamespacedKey WOLF_AVOID_TARGET = NamespacedKey.minecraft("wolf_avoid_target");
++    NamespacedKey RABBIT_PANIC = NamespacedKey.minecraft("rabbit_panic");
++    NamespacedKey FOLLOW_ENTITY = NamespacedKey.minecraft("follow_entity");
++    NamespacedKey LOOK_AT_PLAYER = NamespacedKey.minecraft("look_at_player");
++    NamespacedKey GHAST_MOVE_TOWARDS_TARGET = NamespacedKey.minecraft("ghast_move_towards_target");
++    NamespacedKey FOLLOW_BOAT = NamespacedKey.minecraft("follow_boat");
++    NamespacedKey PANDA_LOOK_AT_PLAYER = NamespacedKey.minecraft("panda_look_at_player");
++    NamespacedKey FISH_RANDOM_SWIM = NamespacedKey.minecraft("fish_random_swim");
++    NamespacedKey PANIC = NamespacedKey.minecraft("panic");
++    NamespacedKey OCELOT_TEMPT = NamespacedKey.minecraft("ocelot_tempt");
++    NamespacedKey RANDOM_STROLL = NamespacedKey.minecraft("random_stroll");
++    NamespacedKey MOVE_THROUGH_VILLAGE = NamespacedKey.minecraft("move_through_village");
++    NamespacedKey DROWNED_ARROW_ATTACK = NamespacedKey.minecraft("drowned_arrow_attack");
++    NamespacedKey PANDA_J = NamespacedKey.minecraft("panda_j");
++    NamespacedKey ENDERMAN_PICKUP_BLOCK = NamespacedKey.minecraft("enderman_pickup_block");
++    NamespacedKey PANDA_L = NamespacedKey.minecraft("panda_l");
++    NamespacedKey GHAST_IDLE_MOVE = NamespacedKey.minecraft("ghast_idle_move");
++    NamespacedKey PANDA_K = NamespacedKey.minecraft("panda_k");
++    NamespacedKey OCELOT_ATTACK = NamespacedKey.minecraft("ocelot_attack");
++    NamespacedKey PANDA_F = NamespacedKey.minecraft("panda_f");
++    NamespacedKey TURTLE_B = NamespacedKey.minecraft("turtle_b");
```

Comment:
Not sure about all these _F etc. fields, are they obfuscated names or something? I feel like it would be useful to have better descriptions of what the goal actually does here.

Reply:
that entire class is auto generated by the junit test. I just scan the classpath for vanilla goals and expose them.
Yes, not all have nice names, but its better than nothing and the easiest way to work with vanilla goals.

Reply:
don't think it's a good idea to expose non-mapped things to api

Reply:
Well, the issue is that you might still want to remove a non mapped vanilla goal.

Reply:
we should give them a field name at least

Reply:
added a obfuscation map to address this